components/restapi/core/component.go

```diff
@@ -152,7 +152,7 @@ func configure() error {
 		}
 
 		return responseByHeader(c, resp)
-	})
+	}, checkNodeSynced())
```

Comment:
Not all endpoints need the node to be synced.
Values that never change after they are created can safely returned by an unsync node.
For example "UTXO changes by slot" are no problem. Either the unsynced node has them or not. No need to be synced. But for transaction metadata or block metadata endpoints, the provided information could be outdated if the node is unsync.

Reply:
I see, I left the ones added for OutputMetadata